source-code/core/plugin/src/resolvePlugins.ts

```diff
@@ -0,0 +1,166 @@
+import { Plugin, ResolvePluginsFunction } from "./api.js"
+import {
+	PluginAppSpecificApiReturnError,
+	PluginFunctionDetectLanguageTagsAlreadyDefinedError,
+	PluginFunctionLoadMessagesAlreadyDefinedError,
+	PluginFunctionSaveMessagesAlreadyDefinedError,
+	PluginUsesInvalidIdError,
+	PluginUsesInvalidSchemaError,
+	PluginUsesReservedNamespaceError,
+} from "./errors.js"
+import { deepmerge } from "deepmerge-ts"
+import { TypeCompiler } from "@sinclair/typebox/compiler"
+
+const whitelistedPlugins = ["inlang.plugin-json", "inlang.plugin-i18next"]
```

Comment:
I don't think this is a smart idea ^^

Reply:
How else would you reserve the `inlang` namespace?

Reply:
We will probably have more issues if we restrict the packages than if we trust the community to not use the inlang namespace.
If we always need to add a new plugin id to this array we
1. will probably forget it every second time and need to release a fix immediately afterwards
2. we require applications to always have the latest version of `@inlang/plugin` installed or else they won't be able to use the newly released plugin